stdlib/public/runtime/Errors.cpp

```diff
@@ -274,39 +299,11 @@ reportNow(uint32_t flags, const char *message)
 #endif
 }
 
-LLVM_ATTRIBUTE_NOINLINE SWIFT_RUNTIME_EXPORT
+SWIFT_ATTRIBUTE_OPTNONE SWIFT_RUNTIME_EXPORT
 void _swift_runtime_on_report(uintptr_t flags, const char *message,
                               RuntimeErrorDetails *details) {
   // Do nothing. This function is meant to be used by the debugger.
-
-  // The following is necessary to avoid calls from being optimized out.
-  asm volatile("" // Do nothing.
-               : // Output list, empty.
-               : "r" (flags), "r" (message), "r" (details) // Input list.
-               : // Clobber list, empty.
-               );
-}
-
-void swift::_swift_reportToDebugger(uintptr_t flags, const char *message,
-                                    RuntimeErrorDetails *details) {
-  _swift_runtime_on_report(flags, message, details);
-}
-
-bool swift::_swift_reportFatalErrorsToDebugger = true;
-
-bool swift::_swift_shouldReportFatalErrorsToDebugger() {
-  return _swift_reportFatalErrorsToDebugger;
-}
-
-/// Report a fatal error to system console, stderr, and crash logs.
-/// Does not crash by itself.
-void swift::swift_reportError(uint32_t flags,
-                              const char *message) {
-#if defined(__APPLE__) && NDEBUG
-  flags &= ~FatalErrorFlags::ReportBacktrace;
-#endif
-  reportNow(flags, message);
-  reportOnCrash(flags, message);
+  SWIFT_OPTNONE_BODY;
```

Comment:
I'm pretty sure I've tried variations of this before and it didn't work: The optimizer was still too smart and it passed `undef` into the arguments (because they're unused), which is why we ended up with the `asm volatile` + input list. Are you 100% sure this actually preserves the arguments?

Reply:
With the `optnone` attribute? That should block all IPO. At least in a small test case it seems to work: https://godbolt.org/g/bUorwg I can leave it as is if you prefer.